lxd/ubuntu-pro/watcher.go

```diff
@@ -0,0 +1,252 @@
+	// The value of `GET /1.0/ubuntu-pro` over devlxd should return the value of the `ubuntu_pro.guest_attach`
+	// configuration of the instance.
+	GuestAttachSettingAvailable GuestAttachSetting = "available"
+
+	// GuestAttachSettingOn indicates that guest attachment is on and can be opted into via guest config. The value of
+	// `GET /1.0/ubuntu-pro` over devlxd should return the value of the `ubuntu_pro.guest_attach` configuration of the
+	// instance.
+	//
+	// NOTE: This setting currently behaves in the same way as GuestAttachSettingAvailable, but we will use this setting
+	// to decide whether automatically set the value of `ubuntu_pro.guest_attach` on instances based on the origin of
+	// their base image when an authenticated simplestreams remote for pro images becomes available.
+	GuestAttachSettingOn GuestAttachSetting = "on"
+)
+
+// isValid returns an error if the GuestAttachSetting is not one of the pre-defined values.
+func (g GuestAttachSetting) isValid() error {
+	if !shared.ValueInSlice(g, []GuestAttachSetting{GuestAttachSettingOff, GuestAttachSettingAvailable, GuestAttachSettingOn}) {
+		return fmt.Errorf("Invalid guest auto-attach setting %q", g)
+	}
+
+	return nil
+}
+
+// ubuntuAdvantageDirectory is the base directory for ubuntu-pro related configuration.
+const ubuntuAdvantageDirectory = "/var/lib/ubuntu-advantage"
+
+// interfacesDir returns the path to the interfaces directory based on the given ubuntu advantage directory.
+func interfacesDir(ubuntuAdvantageDir string) string {
+	return filepath.Join(ubuntuAdvantageDir, "interfaces")
+}
+
+// lxdConfigFile returns the path to the ubuntu-pro LXD configuration file based on the given "interfaces" directory.
+func lxdConfigFile(interfacesDir string) string {
+	return filepath.Join(interfacesDir, "lxd-config.json")
+}
+
+// Settings contains methods for returning ubuntu-pro related settings.
+type Settings interface {
+	GuestAttach() GuestAttachSetting
+}
+
+// proNotEnabled is returned from New if the ubuntuAdvantageDirectory does not exist on the host.
+type proNotEnabled struct{}
+
+// GuestAttach always returns GuestAttachSettingOff.
+func (proNotEnabled) GuestAttach() GuestAttachSetting {
+	return GuestAttachSettingOff
+}
+
+// proWatcher implements the Settings interface with a file watcher.
+type proWatcher struct {
+	watcher            *fsnotify.Watcher
+	guestAttachSetting GuestAttachSetting
+}
+
+// GuestAttach returns the current GuestAttachSetting from the ubuntu-pro configuration.
+func (p *proWatcher) GuestAttach() GuestAttachSetting {
+	return p.guestAttachSetting
+}
+
+// SettingsFile contains ubuntu-pro settings relevant to LXD.
+type SettingsFile struct {
+	GuestAttach GuestAttachSetting `json:"guest_attach"`
+}
+
+// New returns the proWatcher implementation of Settings using a file watcher. When the given context is cancelled, all file
+// watchers are closed and GuestAttach will return GuestAttachSettingOff.
+func New(ctx context.Context) Settings {
```

Comment:
Suggest having this return a point to an `ubuntupro.State` struct that has `GuestAttachSetting()` function on it.

Reply:
or just a `pro` struct, which we can add to in the future, and have a function called `GuestAttachState()`.